### PR TITLE
feat: Guardians can propose, cast votes and execute a freeze/key rotation operation through their view

### PR DESCRIPTION
**Pull request by giovanni-orciuolo**
The flow goes like this: when a wallet is multisig, a guardian can propose an operation. When a proposal is made, the guardian who proposed it automatically casts a "yes" vote for it. The other guardians can then come to the page and vote for the proposal ("yes", "no" or "abstain"). If the proposal reaches a number of positive votes greater than or equal to the defined threshold for the wallet, then the proposal can be executed. This is done manually by one of the guardians, it doesn't matter who does it, by clicking on the passed proposal in the view. It is always possible to see the status of a proposal, and the list of casted votes.

### CODE REVIEW

**Review by j0nl1 — CHANGES_REQUESTED** [review]
I think in general we could reduce complexity, doing components smaller.

For example the function groupVoteListByVote there is no need to live in ProposalDetailsModal, it will make the component more simple and also it don't have to re-render that function anymore. 

If a component is re-render all the content inside the component are rendering included functions, not just html nodes.

Another example will be HTML complexity, in the same component we have these nodes:
```jsx
<div className="flex items-center w-full justify-around mb-2">
                      <p className="text-green-600">YES: {groupVoteListByVote(voteList, "yes").weight}</p>
                      <p className="text-red-600">NO: {groupVoteListByVote(voteList, "no").weight}</p>
                      <p className="text-gray-600">ABSTAIN: {groupVoteListByVote(voteList, "abstain").weight}</p>
                    </div>
```

I think that could be a small component which receive a voteList.

These suggestions don't have to be changed now, just to keep it in mind for the future.

File: services/vectis.ts [hunk 1]
```diff
@@ -163,23 +163,33 @@ export async function toggleProxyWalletFreezeStatus(
...
   signingClient: SigningCosmWasmClient,
   guardianAddress: string,
-  proxyWalletMultisigAddress: string,
+  proxyWalletAddress: string,
+  multisigAddress: string,
   operation: "TOGGLE_FREEZE" | "ROTATE_KEY",
   newUserAddress?: string
 ) {
-  let message = {};
+  let title = "",
```
[the remark this answers is not in the document]
Comment (giovanni-orciuolo) [reply services/vectis.ts]: Done in [`3102eca` (#22)](https://github.com/nymlab/vectis_fe/pull/22/commits/3102ecaca3ca9656c41e40b80d445f6df44fa57b)

[the remark this answers is not in the document]
Comment (j0nl1) [reply services/vectis.ts]: awesome!! 🙌

**Comment by giovanni-orciuolo** [comment]
@j0nl1 In [`67a4d32` (#22)](https://github.com/nymlab/vectis_fe/pull/22/commits/67a4d3243aa2eb17c674315f94918b63c15a68c7) I made a Modal component which we can reuse to create a modal. I plan to later separate the modal children into separate components so we can use them without them being in a modal (more separation of concerns)

File: components/modals/ProposalDetailsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            {loading && <Loader>Fetching proposal votes...</Loader>}
+            {!loading && error && (
+              <div className="my-5">
+                <AlertError>{error.message}</AlertError>
+              </div>
+            )}
+            {!loading && !error && (
+              <>
+                <h3 className="text-2xl font-bold my-5">Votes</h3>
+                {voteList?.length > 0 && (
```
[the remark this answers is not in the document]
Comment (giovanni-orciuolo) [reply components/modals/ProposalDetailsModal.tsx]: Done in [`31933d3` (#22)](https://github.com/nymlab/vectis_fe/pull/22/commits/31933d32845fcac77487052ff137b1822dcef42d)

File: components/buttons/FreezeButton.tsx [hunk 3]
```diff
@@ -60,18 +85,40 @@ export default function FreezeButton({
...
-        className="btn btn-primary btn-md hover:text-base-100 text-xl rounded-full flex-grow mx-2"
-        onClick={proposeToggleFreezeStatus}
-      >
-        PROPOSE FREEZE
-      </button>
+      <>
+        <label
+          htmlFor={freezeProposal ? `vote-modal-${freezeProposal.id}` : ""}
+          className={`btn btn-md hover:text-base-100 text-xl rounded-full flex-grow mx-2 ${
+            alreadyVoted ? "btn-disabled" : "btn-primary"
```
[the remark this answers is not in the document]
Comment (giovanni-orciuolo) [reply components/buttons/FreezeButton.tsx:108]: This check is already present in the `openVoteModal()` functions https://github.com/nymlab/vectis_fe/blob/feat/multisig-voting/components/buttons/FreezeButton.tsx#L91 (same in `RotateKeyButton`)

File: components/buttons/RotateKeyButton.tsx [hunk 4]
```diff
@@ -1,15 +1,86 @@
...
+    setVoteModalOpen(true);
+  }
+
+  if (proxyWalletInfo.multisig_address) {
+    return (
+      <>
+        <label
+          htmlFor={keyRotationProposal ? `vote-modal-${keyRotationProposal.id}` : "rotate-key-modal"}
+          className={`btn btn-md hover:text-base-100 text-xl rounded-full flex-grow mx-2 ${
+            alreadyVoted ? "btn-disabled" : "btn-primary"
```
[the remark this answers is not in the document]
Comment (giovanni-orciuolo) [reply components/buttons/RotateKeyButton.tsx:58]: This check is already present in the `openVoteModal()` functions https://github.com/nymlab/vectis_fe/blob/feat/multisig-voting/components/buttons/FreezeButton.tsx#L91 (same in `RotateKeyButton`)

File: components/GuardianView.tsx [hunk 5]
```diff
@@ -141,21 +153,44 @@ export default function GuardianView() {
             <p>
               Balance: <TokenAmount token={walletInfo.balance} />
             </p>
-            {walletHasFreezeProposal() && <p>Proposal to freeze wallet is active</p>}
-            {walletHasKeyRotationProposal() && <p>Proposal to rotate owner key is active</p>}
+            {getWalletFreezeProposal() && (
```
[the remark this answers is not in the document]
Comment (giovanni-orciuolo) [reply components/GuardianView.tsx]: Done in [`6f3fbde` (#22)](https://github.com/nymlab/vectis_fe/pull/22/commits/6f3fbde032bcef6bea547e08632b9f06cf57291a)

**Comment by whalelephant** [comment]
I have tested the workflow it woks well.

**Review by whalelephant — APPROVED** [review]

**Review by j0nl1 — APPROVED** [review]

**Pull request merged**